solutions/usaco-760.mdx

```diff
@@ -118,3 +119,36 @@ int main(){
 
 </CPPSection>
 </LanguageSection>
+
+<LanguageSection>
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+<PySection>
+
+```py
+data = open('shuffle.in').read().strip().split('\n')
+n = int(data[0])
+moves = list(map(int, data[1].split(' ')))
+order = data[2].split(' ')
+
+for i in range(3):
+    # Create a temp array to hold the order one shuffle back in time
+    temp = [''] * n
+    for j in range(n):
+        temp[j] = order[moves[j] - 1]
+    # Set this temporary array to the order of cows
+    order = temp
+
+ans = ''
+for i in range(n):
+    if i != n - 1:
+        ans += order[i] + '\n'
+    else:
+        ans += order[i]
+
+print(ans, file = open('shuffle.out', 'w'))
```

Comment:
wait no don't put spaces between default arguments?

Reply:
oh oops, sorry about that.